stl/inc/xloctime

```diff
@@ -578,6 +578,9 @@ protected:
             if (*_Fmtfirst == '%') {
                 _First = do_get(_First, _Last, _Iosbase, _State, _Pt,
                     *++_Fmtfirst); // convert a single field
+            } else if (_First == _Last) {
+                _State |= ios_base::failbit;
+                break;

```

Comment:
I meant like this, so that the loop will exit as soon as eof is detected:
```suggestion
            if (_First == _Last) {
                _State = ios_base::failbit | ios_base::eofbit;
                return;
            } else if (*_Fmtfirst == '%') {
                _First = do_get(_First, _Last, _Iosbase, _State, _Pt,
                    *++_Fmtfirst); // convert a single field
```

Reply:
i think it is better to have one return and `if (_First == _Last`) after setting `ios_base::failbit` use break instead of `retrun;`